pephub/routers/eido/eido.py

```diff
@@ -377,28 +113,57 @@ async def validate(
 
             p = peppy.Project(f"{dirpath}/{init_file.filename}")
 
-    # save schema string to temp file, then read in with eido
-    with tempfile.NamedTemporaryFile(mode="w") as schema_file:
-        schema_file.write(schema)
-        schema_file.flush()
+    if schema is None and schema_registry is None:
+        raise HTTPException(
+            status_code=400,
+            detail={
+                "error": "Must supply either a registry path or a list of files to validate."
+            },
+        )
+    
+    if schema_registry is not None:
+        schema_url = f"https://schema.databio.org/{schema_registry}.yaml"
+            
         try:
-            schema_dict = eido.read_schema(schema_file.name)[0]
-        except eido.exceptions.EidoSchemaInvalidError as e:
+            response = requests.get(schema_url)
+            response.raise_for_status()  # Check if the request was successful
+            yaml_string = response.text
+        except requests.exceptions.RequestException as e:
             raise HTTPException(
-                status_code=200,
-                detail={"error": f"Schema is invalid: {str(e)}"},
+                status_code=400,
+                detail={"error": f"Error fetching schema: {str(e)}"},
             )
 
-    # validate
+        # save schema string to temp file
+        with tempfile.NamedTemporaryFile(mode='w', delete=False) as yaml_file:
+            yaml_file.write(yaml_string)
+            schema_dict = yaml_file.name
+    else:
+        # save schema string to temp file, then read in with eido
+        with tempfile.NamedTemporaryFile(mode="w") as schema_file:
+            schema_file.write(schema)
+            schema_file.flush()
+            try:
+                schema_dict = eido.read_schema(schema_file.name)[0]
+            except eido.exceptions.EidoSchemaInvalidError as e:
+                raise HTTPException(
+                    status_code=200,
+                    detail={"error": f"Schema is invalid: {str(e)}"},
+                )
+            
+    # validate project
     try:
         eido.validate_project(
             p,
             schema_dict,
         )
+
     # while we catch this, its still a 200 response since we want to
     # return the validation errors
     except eido.exceptions.EidoValidationError as e:
-        return {"valid": False, "errors": str(e)}
+        errors = ["-> " + str(error) for error in e.errors_by_type]
+        error_message = "\n".join(errors)
+        return {"valid": False, "errors": {str(error_message)}}
```

Comment:
Do we want to return a list of errors instead? that way we can iterate over them on the client nicely.

Reply:
We're returning a list like this
![image](https://github.com/pepkit/pephub/assets/17304717/a18d5b67-2453-4551-b1c8-173fc6439c10)
Does this look ok?

Reply:
I like it:)

Reply:
I think it would be nice to print sample names, where required property doesn't exist. If `sample_name == 'project' it means this property was required in project not in sample`.
But I think right now it looks good